File: lib/core/src/main/java/io/loglens/core/MemorySink.java
package io.loglens.core;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/** 테스트용. 찍힌 줄을 그대로 들고 있는다. */
public final class MemorySink implements Sink {

    private final List<String> lines = Collections.synchronizedList(new ArrayList<>());
    private final List<String> bodies = Collections.synchronizedList(new ArrayList<>());

    @Override
    public void write(Level level, String tag, String body) {
        bodies.add(body);
        lines.add(LineFormat.render(level, tag, body));
    }

    /** 접두사 포함 완성된 줄들. */
    public List<String> lines() { return new ArrayList<>(lines); }

    /** 본문만 (계약의 핵심 부분). */
    public List<String> bodies() { return new ArrayList<>(bodies); }

    public String lastBody() { return bodies.isEmpty() ? null : bodies.get(bodies.size() - 1); }

    public void clear() { lines.clear(); bodies.clear(); }

    public int size() { return bodies.size(); }
}

File: lib/core/src/main/kotlin/io/loglens/core/MemorySink.kt
package io.loglens.core

import java.util.Collections

/** 테스트용. 찍힌 줄을 그대로 들고 있습니다. */
class MemorySink : Sink {

    private val _lines = Collections.synchronizedList(mutableListOf<String>())
    private val _bodies = Collections.synchronizedList(mutableListOf<String>())

    override fun write(level: Level, tag: String, body: String) {
        _bodies.add(body)
        _lines.add(LineFormat.render(level, tag, body))
    }

    /** 줄 앞부분까지 붙은 완성된 줄들. */
    fun lines(): List<String> = ArrayList(_lines)

    /** 본문만 (형식의 핵심 부분). */
    fun bodies(): List<String> = ArrayList(_bodies)

    fun lastBody(): String? = _bodies.lastOrNull()

    fun clear() {
        _lines.clear()
        _bodies.clear()
    }

    val size: Int get() = _bodies.size
}
